tests/v2/integration/catalogv2/system_charts_test.go

```diff
@@ -0,0 +1,439 @@
+	w.Require().NoError(w.updateSetting("system-managed-charts-operation-timeout", "300s"))
+	w.Require().NoError(w.updateSetting("system-feature-chart-refresh-seconds", "21600"))
+}
+
+func (w *SystemChartsTest) SetupSuite() {
+	var err error
+	testSession := session.NewSession()
+	w.session = testSession
+	w.client, err = rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+	insecure := true
+	w.client.RancherConfig.Insecure = &insecure
+	w.catalogClient, err = w.client.GetClusterCatalogClient("local")
+	require.NoError(w.T(), err)
+
+	kubeConfig, err := kubeconfig.GetKubeconfig(w.client, "local")
+	require.NoError(w.T(), err)
+
+	restConfig, err := (*kubeConfig).ClientConfig()
+	require.NoError(w.T(), err)
+
+	//Use this if running locally
+	//restConfig.Insecure = true
+
+	w.restClientGetter, err = kubeconfig.NewRestGetter(restConfig, *kubeConfig)
+	require.NoError(w.T(), err)
+	c, err := w.client.Management.Cluster.ByID("local")
+	require.NoError(w.T(), err)
+	w.cluster = c
+	w.Require().NoError(w.updateSetting("system-managed-charts-operation-timeout", "45s"))
+}
+
+func (w *SystemChartsTest) resetSettings() {
+	w.T().Helper()
+	w.Require().NoError(w.resetManagementCluster())
+	w.Require().NoError(w.uninstallApp("cattle-system", "rancher-aks-operator"))
+	w.Require().NoError(w.uninstallApp("cattle-system", "rancher-aks-operator-crd"))
+	clusterRepo, err := w.catalogClient.ClusterRepos().Get(context.TODO(), "rancher-charts", metav1.GetOptions{})
+	w.Require().NoError(err)
+	if clusterRepo.Spec.GitRepo != "https://github.com/rancher/charts" || clusterRepo.Spec.GitBranch != "release-v2.7" {
+		clusterRepo.Spec.GitRepo = "https://github.com/rancher/charts"
+		clusterRepo.Spec.GitBranch = "release-v2.7"
+		clusterRepo, err = w.catalogClient.ClusterRepos().Update(context.TODO(), clusterRepo, metav1.UpdateOptions{})
+		w.Require().NoError(err)
+		downloadTime := clusterRepo.Status.DownloadTime
+		w.Require().NoError(w.pollUntilDownloaded("rancher-charts", downloadTime))
+	}
+}
+
+func TestOperatorChartsSuite(t *testing.T) {
+	suite.Run(t, new(SystemChartsTest))
+}
+
+func (w *SystemChartsTest) TestInstallChartLatestVersion() {
+	defer w.resetSettings()
+
+	w.Require().NoError(w.updateManagementCluster())
+	app, err := w.waitForAksChart(rv1.StatusDeployed, "rancher-aks-operator", 0)
+	w.Require().NoError(err)
+	latest, err := w.catalogClient.GetLatestChartVersion("rancher-aks-operator")
+	w.Require().NoError(err)
+	w.Assert().Equal(app.Spec.Chart.Metadata.Version, latest)
+}
+
+func (w *SystemChartsTest) TestUpgradeChartToLatestVersion() {
+	defer w.resetSettings()
+
+	clusterRepo, err := w.catalogClient.ClusterRepos().Get(context.TODO(), "rancher-charts", metav1.GetOptions{})
+	w.Require().NoError(err)
+	origClusterRepo := clusterRepo.DeepCopy()
+	cfg, err := w.client.Steve.SteveType(configmaps.ConfigMapSteveType).ByID(fmt.Sprintf("%s/%s", origClusterRepo.Status.IndexConfigMapNamespace, origClusterRepo.Status.IndexConfigMapName))
```

Comment:
Instead of steve client, can we use the client like catalogClient to fetch configmap ?

Reply:
There is a Kubernetes client sigs.k8s.io/controller-runtime/pkg/client which you can use to get config maps